src/erlcloud_kinesis.erl

```diff
@@ -560,3 +563,100 @@ split_shards(StreamName, ShardToSplit, NewStartingHashKey) ->
 split_shards(StreamName, ShardToSplit, NewStartingHashKey, Config) when is_record(Config, aws_config) ->
   Json = [{<<"StreamName">>, StreamName}, {<<"ShardToSplit">>, ShardToSplit}, {<<"NewStartingHashKey">>, NewStartingHashKey}],
   erlcloud_kinesis_impl:request(Config, "Kinesis_20131202.SplitShard", Json).
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Kinesis API: AddTagsToStream
+%% Adds or updates tags for the specified Amazon Kinesis stream.
+%%
+%% erlcloud_kinesis:add_tags(<<"stream_name">>,
+%%                           [{<<"tag_key">>, <<"tag_value">>}]).
+%% ok.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec add_tags(binary(), [{binary(), binary()}]) -> ok | {error, any()}.
+add_tags(StreamName, Tags) ->
+    add_tags(StreamName, Tags, default_config()).
+
+-spec add_tags(binary(), [{binary(), binary()}], aws_config()) ->
+    ok | {error, any()}.
+add_tags(StreamName, Tags, Config) when is_record(Config, aws_config) ->
+    Json      = [{<<"StreamName">>, StreamName},
+                 {<<"Tags">>,       Tags}],
+    Operation = "Kinesis_20131202.AddTagsToStream",
+    Response  = erlcloud_kinesis_impl:request(Config, Operation, Json),
+    normalize_tags_response(Response).
+
+%%------------------------------------------------------------------------------
+%% @doc
+%% Kinesis API: ListTagsForStream
+%% Lists the tags for the specified Amazon Kinesis stream.
+%%
+%% erlcloud_kinesis:list_tags(<<"stream_name">>, <<"key1">>, 1).
+%% {ok, [
+%%       {<<"HasMoreTags">>, true},
+%%       {<<"Tags">>,        [[{<<"Key">>,   <<"key2">>},
+%%                             {<<"Value">>, <<"value">>}]]}
+%%      ]
+%% }.
+%%
+%% @end
+%%------------------------------------------------------------------------------
+
+-spec list_tags(binary()) -> {ok, proplist()} | {error, any()}.
```

Comment:
Agree, `list_all_tags` should be the right solution here.
Regarding previous comment, I don't think we have to force changing AWS response, and `HasMoreTags` has its practical value in some use cases. In case of `list_tags` we just return what we get from AWS.